types/serialize_types.go

```diff
@@ -199,6 +200,19 @@ func SerializeHashType(hashType ScriptHashType) (string, error) {
 	return "", errors.New("Invalid script hash_type: " + string(hashType))
 }
 
+func SerializeHashTypeByte(hashType ScriptHashType) (byte, error) {
+	switch hashType {
+	case HashTypeData:
+		return 0x00, nil
+	case HashTypeType:
+		return 0x01, nil
+	case HashTypeData1:
+		return 0x02, nil
+	default:
```

Comment:
I think we should not return the default enum var with error for code robustness
```suggestion
	default:
	    return 0xFF, errors.New("unknown hash type " + string(hashType))
```

Reply:
The convention in Go to return value together with non-empty error is the default value, like 0 for number, false for boolean and nil for pointer. I think returning 0 follows this convention.